core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -180,6 +180,11 @@ class DAGScheduler(
   /** If enabled, FetchFailed will not cause stage retry, in order to surface the problem. */
   private val disallowStageRetryForTest = sc.getConf.getBoolean("spark.test.noStageRetry", false)
 
+  /** Number of consecutive fetch failures allowed before a stage is aborted */
+  private[scheduler] val maxConsecutiveFetchFailuresPerStage =
+    sc.getConf.getInt("spark.max.fetch.failures.per.stage",
```

Comment:
I don't like this name, it is verbose and it sounds like you might give up on a stage if one *attempt* has too many fetch failures.

I also hate naming though :p

how about "spark.stage.maxAttempts"?  Sort of parallel to "spark.task.maxFailures".